apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore.Snippets/LoggingSnippets.cs

```diff
@@ -0,0 +1,180 @@
+        }
+
+        public void Dispose()
+        {
+            _client.Dispose();
+            _server.Dispose();
+        }
+    }
+
+    /// <summary>
+    /// A simple web application to test the <see cref="GoogleLogger"/>
+    /// and associated classes.
+    /// </summary>
+    internal class LoggingTestApplication
+    {
+        private static readonly string ProjectId = Utils.GetProjectIdFromEnvironment();
+
+        // Sample: RegisterGoogleLogger
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddMvc();
+        }
+
+        public void Configure(IApplicationBuilder app, ILoggerFactory loggerFactory)
+        {
+            // Replace ProjectId with your Google Cloud Project ID.
+            loggerFactory.AddGoogle(app.ApplicationServices, ProjectId);
+
+            // ...any other configuration your application requires.
+
+            app.UseMvc(routes =>
+            {
+                routes.MapRoute(
+                    name: "default",
+                    template: "{controller=Home}/{action=Index}/{id?}");
+            });
+        }
+        // End sample
+    }
+
+    /// <summary>
+    /// A simple web application to test the <see cref="GoogleLogger"/>
+    /// and associated classes in ASP.Net Core 2.
+    /// </summary>
+    internal class LoggingTestApplicationCore2
+    {
+        private static readonly string ProjectId = Utils.GetProjectIdFromEnvironment();
+
+        // Sample: RegisterGoogleLogger2
+        public void ConfigureServices(IServiceCollection services)
+        {
+            // Replace ProjectId with your Google Cloud Project ID.
+            services.AddSingleton<ILoggerProvider>(sp => GoogleLoggerProvider.Create(sp, ProjectId));
+
+            services.AddMvc();
+        }
+
+        public void Configure(IApplicationBuilder app)
+        {
+            // ...any configuration your application requires.
+
+            app.UseMvc(routes =>
+            {
+                routes.MapRoute(
+                    name: "default",
+                    template: "{controller=Home}/{action=Index}/{id?}");
+            });
+        }
+        // End sample
+    }
+
+    /// <summary>
+    /// A controller for the <see cref="LoggingTestApplication"/>
+    /// or for the <see cref="LoggingTestApplicationCore2"> used to log information.
+    /// </summary>
+    public class LoggingSamplesController : Controller
+    {
+        internal const string LoggerName = "Logger for Testing";
+
+        // Sample: UseGoogleLogger
+        /// <summary>
+        /// This method logs an Information level message.
+        /// The <see cref="ILoggerFactory"/> is populated by dependency injection
+        /// thanks to the use of the <see cref="FromServicesAttribute"/> attribute.
+        /// </summary>
+        public void LogInformation(string id, [FromServices]ILoggerFactory loggerFactory)
```

Comment:
I *think* we can just pass in an ILogger here and it will be created based on the class name

Reply:
OK, I'll change that, it seems simpler. Maybe at some point we should consider adding more snippets to this, explicitly naming the logger might have it's use cases.

Reply:
I think more snippets is always better 👍